### PR TITLE
cql-pytest: add instructions on how to get Cassandra

### PR DESCRIPTION
**Pull request by nyh**
The cql-pytest framework allows running tests also against Cassandra, but developers need to install Cassandra on their own because modern distributions such as Fedora no longer carry a Cassandra package.

This patch adds clear and easy to follow (I think) instructions on how to download a pre-compiled Cassadra, or alternatively how to download and build Cassandra from source - and how either can be used with the test/cql-pytest/run-cassandra script.

### CODE REVIEW

**Comment by nyh** [comment]
@nodep recently I explained to you how to build Cassandra, but downloading an already built Cassandra is even easier - and here I include instructions how how to do either. Both instructions should be very easy to follow (I believe) to follow, but you are welcome to review them.

@kostja  this is what a developer needs to do to be able to run Cassandra. Yes, I know, there is also a docker alternative. Maybe one day I'll add support to it in test/cql-pytest/run-cassandra (should be easy). But although docker is easier to start up than installing "java-11" and a precompiled Cassandra (you can run your first Cassandra in 2 minutes, instead of 3 with my method), it is actually harder to understand how to later run and configure Cassandra. The `test/cql-pyest/run-cassandra` scripts avoids the need to learn how to run Cassandra - it runs Cassandra for you on a temporary directory and all the relevant configuration parameters, and deletes everything when the test is over.

File: test/cql-pytest/README.md [hunk 1]
```diff
@@ -202,3 +206,71 @@ and practices be followed when writing new tests:
...
+install it manually, and this section explains how. It's very easy, and
+don't worry - you don't even need to learn how to run Cassandra, as the
+"test/cql-pytest/run-cassandra" tool will do it for you.
+
+To be able to run Cassandra, you'll need either Java 8 or 11 installed on
+your system - Cassandra does not support more recent versions of Java.
+However, this old Java only needs to be installed *alongside* your favorite
+version of Java - it does not need to be the default Java on your system.
+The "run-cassandra" script will automatically pick the right version of
+Java from multiple versions installed on your system. Of modern Fedora,
```
Comment (nodep) [inline test/cql-pytest/README.md:227]: Small typo; should say "_On_ modern Fedora" instead of "_Of_ modern Fedora"

**Review by nodep — APPROVED** [review]
Looks good

File: test/cql-pytest/README.md [hunk 2]
```diff
@@ -202,3 +206,71 @@ and practices be followed when writing new tests:
...
+installing Java 11 as a secondary Java is as simple as `dnf install java-11`.
```
Comment (ptrsmrn) [inline test/cql-pytest/README.md:228]:
```suggestion
installing Java 11 as a secondary Java is as simple as `sudo dnf install java-11`.
```

File: test/cql-pytest/README.md [hunk 3]
```diff
@@ -202,3 +206,71 @@ and practices be followed when writing new tests:
...
+export CASSANDRA=/tmp/apache-cassandra-4.1.4/bin/cassandra
+test/cql-pytest/run-cassandra testfile.py::testfunc
+```
+
+## Building Cassandra from source code
+Usually, installing a pre-compiled Cassandra is enough. But in some cases
+you might want to test some unofficial or modified version of Cassandra,
+built from source. This is also not difficult:
+
+First, download the Cassandra source code, e.g., from github:
```
Comment (ptrsmrn) [inline test/cql-pytest/README.md:252]:
```suggestion
First, download the Cassandra source code, e.g. from github:
```

**Comment by mykaul** [comment]
I still feel https://github.com/testcontainers/testcontainers-python/pull/167 is a better path

**Review by ptrsmrn — APPROVED** [review]

File: test/cql-pytest/README.md (hunk 2 above)
Comment (nyh) [reply test/cql-pytest/README.md:228]: Ok, I'll do that change.

File: test/cql-pytest/README.md (hunk 1 above)
Comment (nyh) [reply test/cql-pytest/README.md:227]: Done

File: test/cql-pytest/README.md (hunk 3 above)
Comment (nyh) [reply test/cql-pytest/README.md:252]: According to the manuals of style I read in the past, "e.g." must be followed with a comma so I always do. But I agree that in this case, the reasoning - even if you know Latin - is dubious. I'll remove it.

**Comment by nyh** [comment]
Pushed a new version with the small typos indicated by the reviewers corrected.

**Comment by nyh** [comment]
> I still feel [testcontainers/testcontainers-python#167](https://github.com/testcontainers/testcontainers-python/pull/167) is a better path

I don't understand what this yet-another-container-framework "testcontainers" gets us over just using docker.  I agree it would have been a good option to make test/cql-pytest/run-cassandra be able to use the latest Cassandra via docker (not a "test container" - just normal docker) and then the user doesn't even know or care where Cassandra is coming from. Cassandra has official instructions for doing this: https://cassandra.apache.org/_/quickstart.html.

Yet, even though we can later add this docker-running option to test/cql-pytest/run-cassandra, it doesn't exist yet and the purpose of this patch is to document what does exist. And as I explain in the text added in this patch - not hard to do.

Please remember that as **developers**, we need to always keep the ability to run modified versions (of Scylla but perhaps even of Cassandra), and not only pre-packaged versions from Docker. So while Docker may be a convenience we allow in the future, the ability described in this patch - to run a Cassandra you downloaded or compiled yourself - should not go away.

**Pull request closed without merging**